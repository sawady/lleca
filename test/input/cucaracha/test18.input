

fun f(a : Int) {
  a := a + 1
  putNum(
    a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + (
    a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + (
    a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + (
    a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + ( a + (
    a + a))))))))))))))))))))))))))))))))))))))))
  )
  putChar(10)
}

fun main() {
  f(0)
  f(1)
  f(2)
  f(4)
}
