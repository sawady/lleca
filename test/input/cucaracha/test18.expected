Cons(
  Function(
    f,
    Cons(
      Parameter(
        a,
        Int
      ),
      Nil
    ),
    Unit,
    Cons(
      StmtAssign(
        a,
        ExprAdd(
          ExprVar(
            a
          ),
          ExprConstNum(
            1
          )
        )
      ),
      Cons(
        StmtCall(
          putNum,
          Cons(
            ExprAdd(
              ExprVar(
                a
              ),
              ExprAdd(
                ExprVar(
                  a
                ),
                ExprAdd(
                  ExprVar(
                    a
                  ),
                  ExprAdd(
                    ExprVar(
                      a
                    ),
                    ExprAdd(
                      ExprVar(
                        a
                      ),
                      ExprAdd(
                        ExprVar(
                          a
                        ),
                        ExprAdd(
                          ExprVar(
                            a
                          ),
                          ExprAdd(
                            ExprVar(
                              a
                            ),
                            ExprAdd(
                              ExprVar(
                                a
                              ),
                              ExprAdd(
                                ExprVar(
                                  a
                                ),
                                ExprAdd(
                                  ExprVar(
                                    a
                                  ),
                                  ExprAdd(
                                    ExprVar(
                                      a
                                    ),
                                    ExprAdd(
                                      ExprVar(
                                        a
                                      ),
                                      ExprAdd(
                                        ExprVar(
                                          a
                                        ),
                                        ExprAdd(
                                          ExprVar(
                                            a
                                          ),
                                          ExprAdd(
                                            ExprVar(
                                              a
                                            ),
                                            ExprAdd(
                                              ExprVar(
                                                a
                                              ),
                                              ExprAdd(
                                                ExprVar(
                                                  a
                                                ),
                                                ExprAdd(
                                                  ExprVar(
                                                    a
                                                  ),
                                                  ExprAdd(
                                                    ExprVar(
                                                      a
                                                    ),
                                                    ExprAdd(
                                                      ExprVar(
                                                        a
                                                      ),
                                                      ExprAdd(
                                                        ExprVar(
                                                          a
                                                        ),
                                                        ExprAdd(
                                                          ExprVar(
                                                            a
                                                          ),
                                                          ExprAdd(
                                                            ExprVar(
                                                              a
                                                            ),
                                                            ExprAdd(
                                                              ExprVar(
                                                                a
                                                              ),
                                                              ExprAdd(
                                                                ExprVar(
                                                                  a
                                                                ),
                                                                ExprAdd(
                                                                  ExprVar(
                                                                    a
                                                                  ),
                                                                  ExprAdd(
                                                                    ExprVar(
                                                                      a
                                                                    ),
                                                                    ExprAdd(
                                                                      ExprVar(
                                                                        a
                                                                      ),
                                                                      ExprAdd(
                                                                        ExprVar(
                                                                          a
                                                                        ),
                                                                        ExprAdd(
                                                                          ExprVar(
                                                                            a
                                                                          ),
                                                                          ExprAdd(
                                                                            ExprVar(
                                                                              a
                                                                            ),
                                                                            ExprAdd(
                                                                              ExprVar(
                                                                                a
                                                                              ),
                                                                              ExprAdd(
                                                                                ExprVar(
                                                                                  a
                                                                                ),
                                                                                ExprAdd(
                                                                                  ExprVar(
                                                                                    a
                                                                                  ),
                                                                                  ExprAdd(
                                                                                    ExprVar(
                                                                                      a
                                                                                    ),
                                                                                    ExprAdd(
                                                                                      ExprVar(
                                                                                        a
                                                                                      ),
                                                                                      ExprAdd(
                                                                                        ExprVar(
                                                                                          a
                                                                                        ),
                                                                                        ExprAdd(
                                                                                          ExprVar(
                                                                                            a
                                                                                          ),
                                                                                          ExprAdd(
                                                                                            ExprVar(
                                                                                              a
                                                                                            ),
                                                                                            ExprAdd(
                                                                                              ExprVar(
                                                                                                a
                                                                                              ),
                                                                                              ExprVar(
                                                                                                a
                                                                                              )
                                                                                            )
                                                                                          )
                                                                                        )
                                                                                      )
                                                                                    )
                                                                                  )
                                                                                )
                                                                              )
                                                                            )
                                                                          )
                                                                        )
                                                                      )
                                                                    )
                                                                  )
                                                                )
                                                              )
                                                            )
                                                          )
                                                        )
                                                      )
                                                    )
                                                  )
                                                )
                                              )
                                            )
                                          )
                                        )
                                      )
                                    )
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                )
              )
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                10
              ),
              Nil
            )
          ),
          Nil
        )
      )
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtCall(
          f,
          Cons(
            ExprConstNum(
              0
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            f,
            Cons(
              ExprConstNum(
                1
              ),
              Nil
            )
          ),
          Cons(
            StmtCall(
              f,
              Cons(
                ExprConstNum(
                  2
                ),
                Nil
              )
            ),
            Cons(
              StmtCall(
                f,
                Cons(
                  ExprConstNum(
                    4
                  ),
                  Nil
                )
              ),
              Nil
            )
          )
        )
      )
    ),
    Nil
  )
)
